### PR TITLE
Add test cases

### PR DESCRIPTION
**Pull request by zshijing**
Description
-----------
Add Callback NULL Check test & add included service test& Long Write test & secondary service test. Add macros to disable/enable them.

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_common.c [hunk 1]
```diff
@@ -584,6 +734,9 @@ void IotTestBleHal_CreateServiceA()
 void IotTestBleHal_CreateServiceB()
 {
     prvCreateService( &_xSrvcB );
+#if ENABLE_TC_ADD_INCLUDED_SERVICE
+    prvCreateIncludedService( &_xSrvcB, bletestATTR_INCLUDED_SERVICE );
```
Comment (ravibhagavandas) [inline libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_common.c]: Should included service be secondary service ( service C ) ?

Comment (zshijing) [reply libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_common.c]: An included service can also be a primary service.

File: libraries/abstractions/ble_hal/test/include/iot_test_ble_hal_integration.h [hunk 2]
```diff
@@ -36,7 +36,8 @@
 
 #include "iot_test_ble_hal_common.h"
 
-#define bletestsFAIL_CHAR_VALUE    "fail"
+#define ENABLE_TC_CALLBACK_NULL_CHECK    ( 0 )
```
[the remark this answers is not in the document]
Comment (zshijing) [reply libraries/abstractions/ble_hal/test/include/iot_test_ble_hal_integration.h]: Sure

File: libraries/abstractions/ble_hal/test/include/iot_test_ble_hal_common.h [hunk 3]
```diff
@@ -46,6 +46,12 @@
 #include IOT_UNITY_FIXTURE
 #include IOT_UNITY
 
+#if ( 0 )
```
[the remark this answers is not in the document]
Comment (zshijing) [reply libraries/abstractions/ble_hal/test/include/iot_test_ble_hal_common.h]: Done

**Pull request merged**